src/Compilers/Core/Portable/Compilation/IExpression.cs

```diff
@@ -0,0 +1,776 @@
+﻿using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Semantics
+{
+    /// <summary>
+    /// Represents a C# or VB expression.
+    /// </summary>
+    public interface IExpression : IOperation
+    {
+        /// <summary>
+        /// Result type of the expression.
+        /// </summary>
+        ITypeSymbol ResultType { get; }
+        /// <summary>
+        /// If the expression evaluates to a constant value, the value of the expression, and otherwise null.
+        /// </summary>
+        object ConstantValue { get; }
+    }
+
+    /// <summary>
+    /// Represents a C# or VB method invocation.
+    /// </summary>
+    public interface IInvocation : IExpression
+    {
+        /// <summary>
+        /// Method to be invoked.
+        /// </summary>
+        IMethodSymbol TargetMethod { get; }
+        /// <summary>
+        /// 'This' or 'Me' argument to be supplied to the method.
+        /// </summary>
+        IExpression Instance { get; }
+        /// <summary>
+        /// Kind of the invocation.
+        /// </summary>
+        InvocationKind InvocationKind { get; }
+        /// <summary>
+        /// Arguments of the invocation, excluding the instance argument.
+        /// </summary>
+        ImmutableArray<IArgument> Arguments { get; }
+        /// <summary>
+        /// Find the argument supplied for a given parameter of the target method.
+        /// </summary>
+        /// <param name="parameter">Parameter of the target method.</param>
+        /// <returns>Argument corresponding to the parameter.</returns>
+        IArgument ArgumentMatchingParameter(IParameterSymbol parameter);
+    }
+
+    /// <summary>
+    /// Kinds of invocations.
+    /// </summary>
+    public enum InvocationKind
```

Comment:
Technically, yes. The one case that can't be covered by looking solely at the target method is a non-virtual call to a virtual method (via "base." in C# or "MyBase." in VB). The encoding of "base." vs. "this." in C# is via a SuppressVirtualCalls flag that is available on all expressions, while in VB the encoding of "MyBase." vs. "MyClass." vs. "Me." is via different kinds of bound nodes. Encoding this information at the point of reference to a method is more straightforward for a consumer than encoding it in object references.

So, the InvocationKind enum could be replaced with a "SuppressVirtual" flag on IInvocation (and on method references and property set/get references also). One could also change the type of the Method property to be a method reference, but that would be a step away from the bound trees and involve allocating helper objects for both VB and C#.